### PR TITLE
Added support for pi2

### PR DESCRIPTION
**Pull request by tylerwowen**
I added support for Pi2. Only tested on Pi2. It works on Pi2, but I'm not sure if it is fully working.

### CODE REVIEW

**Comment by metachris** [comment]
Thank you! I will get a Pi 2 soon and then I'll test this PR immediately.

**Comment by tylerwowen** [comment]
You are welcome. However, please notice that the PWM part is still not working on Pi 2. I had difficulties to fix that part.

**Comment by metachris** [comment]
Alright. Perhaps its worth to take a look at the ServoBlaster library, which was the starting point for the RPIO PWM module:

* https://github.com/richardghirst/PiBits/tree/master/ServoBlaster
* https://github.com/richardghirst/PiBits/issues/43

**Comment by metachris** [comment]
Also:

* https://github.com/metachris/RPIO/network
* https://github.com/mw46d/RPIO/commit/a92a859848e3e87f5a6f51614093c2c3cd7d283e

**Comment by fjbelchi** [comment]
Any progress with PWM support for Pi 2?

**Comment by WORMSS** [comment]
Any progress with PWM support for Pi 2? Whole reason I got the Pi2 because I was told it's better than Arduino. But I got the Arduino running a simple PWM within 5 minutes. I've been on this for an hour on the Pi2 trying libraries left right and centre. :bangbang:

**Comment by jashort** [comment]
@WORMSS I had good luck with the PWM support in the fork @tylerwowen made. I installed it from https://github.com/tylerwowen/RPIO (It's this pull request, it just hasn't been merged yet)

**Comment by tylerwowen** [comment]
@WORMSS Raspberry Pi is not better for PWM. This is from the [official GPIO module documentation](https://pypi.python.org/pypi/RPi.GPIO)

> Note that this module is unsuitable for real-time or timing critical applications. This is because you can not predict when Python will be busy garbage collecting. It also runs under the Linux kernel which is not suitable for real time applications - it is multitasking O/S and another process may be given priority over the CPU, causing jitter in your program. If you are after true real-time performance and predictability, buy yourself an Arduino http://www.arduino.cc !

Use the official library instead, it has software PWM supports.

@jashort Did it really work for PWM? I don't think I could fix it.

**Comment by WORMSS** [comment]
@jashort @tylerwowen thank you. I guess I will do a combination of the two then. As I think I will still need certain things that a Pi2 can offer such as nodejs (and ssh if things get really bad). So maybe a Pi2 that triggers an Arduino into action. Fingers crossed. Again, thank  you.

**Comment by Xfel** [comment]
Any progress on this? As far as I see the PWM stuff does work now in ServoBlaster after this commit: https://github.com/richardghirst/PiBits/commit/96014c804d68677baf4fff25178f9b152c1f31e6

So could this be adapted for this project too?

**Comment by metachris** [comment]
I wasn't aware, thanks for the heads up! If you have time to port it over,
you are more than welcome to :-)

**Comment by cmdwiggles** [comment]
Any progress with this?
Thoroughly dissatisfied with GPIO.PWM and its ideas of 'pwm', veeerrrrry jittery outputs dont end well when mixed with drones

**Comment by alexellis** [comment]
@Xfel @metachris I can test any merge that is done from servo blaster.

If the pwm side of things is based upon the work of servoblaster - how difficult would it be to merge? Was there a lot of change to the methods/interface?

**Comment by maleadt** [comment]
Differences between [servoblaster](https://github.com/richardghirst/PiBits/blob/master/ServoBlaster/user/servod.c) and [RPIO](https://github.com/metachris/RPIO/blob/master/source/c_pwm/pwm.c) are significant, so merging isn't a quick and easy job.

I tried porting the PWM part to RPi2, but couldn't get it to work either (it will require more than just updating the peripheral base offset). That said, `servoblaster` didn't really work on my Pi2: initially (first 5 seconds or so) it PWMs fine, but then the board powers-off abruptly. No oops or nothing, so I didn't try to debug. Not sure whether it's specific to my set-up or board though.
So for anybody trying to port the servoblaster Pi2 support over: make sure servoblaster itself works first.

File: source/c_gpio/py_gpio.c~ [hunk 1]
```diff
@@ -0,0 +1,577 @@
+/*
```
Comment (lurch) [inline source/c_gpio/py_gpio.c~]: This is a backup file (ends with ~ ) so should be deleted from this PR.

File: source/c_gpio/cpuinfo.c [hunk 2]
```diff
@@ -70,8 +71,10 @@ get_cpuinfo_revision(char *revision_hex)
     if ((strcmp(revision_hex, "0002") == 0) ||
         (strcmp(revision_hex, "0003") == 0)) {
         return 1;
-    } else if ((strcmp(revision_hex, "0010") == 0)) {
-        // We'll call Model B+ (0010) rev3
+    } else if (strcmp(revision_hex, "0010") == 0
+							 || strcmp(revision_hex, "a21041") == 0
+							 || strcmp(revision_hex, "a01041") == 0 ) {
```
Comment (lurch) [inline source/c_gpio/cpuinfo.c]: Looks like the indentation went a bit funny? ;)

File: source/c_gpio/c_gpio.c [hunk 3]
```diff
@@ -79,8 +100,16 @@ setup(void)
...
-
+    int type = get_CPU_info();
+    if (type == -1) {
+        return SETUP_MMAP_FAIL;
+		}
+    else if (type == 1 ){
+        gpio_map = (uint32_t *)mmap( (void *)gpio_mem, BLOCK_SIZE, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_FIXED, mem_fd, BCM2708_PERI_BASE + 0x200000);
+		}
+    else{
+				gpio_map = (uint32_t *)mmap( (void *)gpio_mem, BLOCK_SIZE, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_FIXED, mem_fd, BCM2709_PERI_BASE + 0x200000);
```
Comment (lurch) [inline source/c_gpio/c_gpio.c]: Strange indentation problems

**Pull request closed without merging**